tests/unit_tests/tools/test_tool.py

```diff
@@ -126,76 +134,56 @@ def test_tool_profiles():
-        result = mock.Mock(returncode=1)
-        mocked_error_message = 'mocked error message'
-        result.stderr.decode = mock.Mock(return_value=mocked_error_message)
-        with mock.patch('fab.tools.tool.subprocess.run',
-                        return_value=result):
-            with pytest.raises(RuntimeError) as err:
-                tool.run()
-            assert mocked_error_message in str(err.value)
-            assert "Command failed with return code 1" in str(err.value)
-
-    def test_error_file_not_found(self):
-        '''Tests the error handling of `run`. '''
-        tool = Tool("does_not_exist", "does_not_exist",
-                    Category.FORTRAN_COMPILER)
-        with mock.patch('fab.tools.tool.subprocess.run',
-                        side_effect=FileNotFoundError("not found")):
-            with pytest.raises(RuntimeError) as err:
-                tool.run()
-            assert ("Command '['does_not_exist']' could not be executed."
-                    in str(err.value))
-
-
-def test_suite_tool():
-    '''Test the constructor.'''
-    tool = CompilerSuiteTool("gnu", "gfortran", "gnu",
-                             Category.FORTRAN_COMPILER)
-    assert str(tool) == "CompilerSuiteTool - gnu: gfortran"
-    assert tool.exec_name == "gfortran"
-    assert tool.name == "gnu"
-    assert tool.suite == "gnu"
-    assert tool.category == Category.FORTRAN_COMPILER
-    assert isinstance(tool.logger, logging.Logger)
+    """
+    Tests tool run method.
+    """
+    def test_no_error_no_args(self, fake_process: FakeProcess) -> None:
+        """
+        Tests run with no aruments.
+        """
+        fake_process.register(['stool'], stdout="123")
+        fake_process.register(['stool'], stdout="123")
+        tool = Tool("some tool", "stool", Category.MISC)
+        assert tool.run(capture_output=True) == "123"
+        assert tool.run(capture_output=False) == ""
+        assert call_list(fake_process) == [['stool'], ['stool']]
+
+    def test_run_with_single_args(self,
+                                  subproc_record: ExtendedRecorder) -> None:
+        """
+        Tets run with single argument.
+        """
+        tool = Tool("some tool", "tool", Category.MISC)
+        tool.run("a")
+        assert subproc_record.invocations() == [['tool', 'a']]
+
+    def test_run_with_multiple_args(self,
+                                    subproc_record: ExtendedRecorder) -> None:
+        """
+        Tests run with multiple arguments.
+        """
+        tool = Tool("some tool", "tool", Category.MISC)
+        tool.run(["a", "b"])
+        assert subproc_record.invocations() == [['tool', 'a', 'b']]
+
+    def test_error(self, fake_process: FakeProcess) -> None:
+        """
+        Tests running a failing tool.
+        """
+        fake_process.register(['tool'], returncode=1)
+        tool = Tool("some tool", "tool", Category.MISC)
+        with raises(RuntimeError) as err:
```

Comment:
This test also changes line covert: tool.s.py line 202 is not tested anymore (appending stdout to the error message). Same as above, I won't block this PR because of this, but it would be nice to have.